code/datums/materials/_material.dm

```diff
@@ -154,7 +154,11 @@ Simple datum which is instanced once per type and is used for every object of sa
 	// We assume no parallax means no space means no light
 	if(SSmapping.level_trait(on.z, ZTRAIT_NOPARALLAX))
 		return
-	on.set_light(2, 0.75, starlight_color)
+	on.set_light(2, 0.75, get_starlight_color())
+
+///Gets the space color and possible changed color if space is different
+/datum/material/proc/get_starlight_color()
+	return (starlight_color == COLOR_STARLIGHT) ? GLOB.starlight_color : starlight_color
```

Comment:
This seems like it's not very future-proofed, what if instead: 

Default the class var `starlight_color` to `null`, and then instead do `return starlight_color || GLOB.starlight_color`
That way it chooses the color (if set) or the global (if not)